lib/tests/test_local_working_copy.rs

```diff
@@ -198,10 +197,32 @@ fn test_checkout_file_transitions(backend: TestRepoBackend) {
     let right_commit = commit_with_tree(&store, right_tree_id.clone());
 
     let ws = &mut test_workspace.workspace;
-    ws.check_out(repo.op_id().clone(), None, &left_commit)
-        .unwrap();
-    ws.check_out(repo.op_id().clone(), None, &right_commit)
-        .unwrap();
+    match ws.check_out(repo.op_id().clone(), None, &left_commit) {
+        Err(CheckoutError::Other { err: outer_err, .. }) => {
+            match outer_err.downcast::<SymlinkError>() {
+                Ok(inner_err)
+                    if matches!(*inner_err, SymlinkError::DeveloperModeNotEnabledError) =>
+                {
+                    return
```

Comment:
Does this mean we skip the entire test case if developer mode is not enabled? I think we still want to check that we can update between symlinks and that snapshotting works without removing the file when developer mode is not enabled.

Reply:
I've added a function in testlib which can check whether we only failed because of missing symlink support and continue. However, this didn't put things in a sufficiently useful state to actually do any partial testing (see below or in discord for more than that) - so I just made it skip the test after all :(

That's better than the status quo - previously the tests were being unconditionally skipped on windows - but I'm not sure how we can have a partially functional test. The checkout process deliberately fails if it isn't able to produce a tree equivalent to what should be in the working copy (and that's a good thing). I think we'd need to break the tests up into smaller units, rather than trying to integration-test a subsystem with an internal failure.